Nested sidebar nav for certification and Front-End Sites section

stevector opened the pull request:
This pull requests starts integrated a nested sidebar element. (I'll capture a fuller plan under https://github.com/pantheon-systems/documentation/issues/9084)

To test the working nested sidebar, go to
* https://pr-9047-documentation.appa.pantheon.site/certification/study-guide
* https://pr-9047-documentation.appa.pantheon.site/guides/decoupled/overview
* And any guide page where this element is not used (to validate those non-nested sidebars didn't break) like https://pr-9047-documentation.appa.pantheon.site/guides/drush

Prior to merge, this PR needs

- [x] Technical review from @mel-miller
- [x] Styling adjustments from @mel-miller (The `navbar` element has styling that should perhaps be moved to the new `OmniSidebarNav` that wraps `navbar`)
- [x] More details for planned follow up written out under https://github.com/pantheon-systems/documentation/issues/9084
- [x] More guidance written into the readme of the new component or the top level readme on how to add adjust the menu items themselves.

stevector commented on prettier.config.js:
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
Maybe I should do changes to prettier in a different PR

stevector commented on package.json:
```diff
@@ -68,7 +68,8 @@
     "prettier": "^2.8.3",
-    "react-date-range": "^1.4.0"
+    "react-date-range": "^1.4.0",
+    "vitest": "^1.6.0"
```
I should add to the readme (or something else) an explanation of vitest usage

stevector commented on src/components/omniSidebarNav/allGuidesQuery.js:
```diff
@@ -0,0 +1,35 @@
+import { useStaticQuery, graphql } from 'gatsby';
+
```
add some comments

stevector commented:
@mel-miller, when you have time, can you review how I've implemented the nested sidebar here?

mel-miller commented:
@stevector -- I don't see any problems with the code and the way that the nav is implemented. 👍 I did have a few small design corrections though. I've indicated them within this screenshot. 

1. For where the red line is along the menu top and right sides, just match the same amount of spacing in the main section (circled in red). 
2. For the external link indicator, it should be after the text instead of before it and use the `externalLink` variant of the [PDS icon component](https://live-pds-toolkit-react.appa.pantheon.site/?path=/docs/components-icon--docs). 

<img width="904" alt="Screenshot 2024-07-16 at 6 29 37 PM" src="https://github.com/user-attachments/assets/e19e652c-c408-45b8-92c6-911421e61e22">

stevector commented:
@mel-miller, thanks! I've added padding and switched the icon. LMK what you think

https://pr-9047-documentation.appa.pantheon.site/certification/study-guide
https://pr-9047-documentation.appa.pantheon.site/guides/decoupled/overview

mel-miller commented:
@stevector -- padding looks good. The icon should be after the text and probably one size smaller.

rachelwhitton commented:
@stevector or @jazzsequence can you resolve merge conflicts with main here when convenient? Conflicts were introduced on this branch after I reverted Prettier on main yesterday, and I'm not familiar enough w/the changes in this PR to resolve

stevector commented:
Yes, though I'll work on getting the Prettier change back into main first.
